packages/components/src/segmented-control/segmented-control.tsx

```diff
@@ -0,0 +1,261 @@
+import {
+  Children,
+  cloneElement,
+  forwardRef,
+  isValidElement,
+  useEffect,
+  useRef,
+  useState,
+} from 'react'
+
+import { cva } from 'cva'
+
+import type { ReactElement, ReactNode } from 'react'
+
+const tabContainerStyles = cva({
+  base: 'relative flex flex-1 items-center justify-center gap-0.5 rounded-10 p-0.5',
+  variants: {
+    type: {
+      grey: 'bg-neutral-10 blur:bg-neutral-80/5 blur:backdrop-blur-[20px] dark:bg-neutral-80 blur:dark:bg-white-5',
+      'dark-grey': 'bg-neutral-20 dark:bg-neutral-90',
+    },
+    size: {
+      '24': 'h-6',
+      '32': 'h-8',
+    },
+  },
+})
+
+const tabStyles = cva({
+  base: 'absolute left-0 flex-1 rounded-8 transition-all duration-300 ease-out',
+  variants: {
+    type: {
+      grey: 'bg-neutral-50 blur:bg-neutral-80/60 dark:bg-neutral-60',
+      'dark-grey': 'bg-neutral-50 dark:bg-neutral-60',
+    },
+    size: {
+      '24': 'h-5',
+      '32': 'h-7',
+    },
+  },
+})
+
+const segmentStyles = cva({
+  base: 'relative z-10 flex flex-1 select-none items-center justify-center whitespace-nowrap rounded-8 font-medium transition-all duration-300 ease-out',
+
+  variants: {
+    active: {
+      true: 'text-white-100',
+      false:
+        'bg-transparent text-neutral-80 hover:bg-neutral-20 dark:text-neutral-40 dark:hover:bg-neutral-70 dark:hover:text-white-100',
```

Comment:
It seems like there's no distinction between `grey` and `dark-grey` regarding hovers.